imagecraft/services/pack.py

```diff
@@ -43,15 +49,75 @@ def __init__(
         self._build_plan = build_plan
 
     @override
-    def pack(self, prime_dir: pathlib.Path, dest: pathlib.Path) -> list[pathlib.Path]:  # noqa: ARG002
+    def pack(self, prime_dir: Path, dest: Path) -> list[Path]:  # noqa: ARG002
         """Pack the image.
 
         :param prime_dir: Directory path to the prime directory.
         :param dest: Directory into which to write the package(s).
         :returns: A list of paths to created packages.
         """
+        # Pydantic has already validated that there is only a single volume before now
+        volume_name, volume = next(iter(cast(Project, self._project).volumes.items()))
+        disk_image_file = dest / (volume_name + os.extsep + "img")
+
+        # Determine necessary image size, and reserve space
+        # "The default start offset for the first partition is 1 MiB", per `man sfdisk`,
+        # plus one more MiB for padding at the end of the disk.
+        image_bytes = diskutil.MIB * 2
+        for structure_item in volume.structure:
+            image_bytes += structure_item.size
+        image_sectors = diskutil.bytes_to_sectors(image_bytes, SECTOR_SIZE)
+        diskutil.create_zero_image(
+            imagepath=disk_image_file,
+            sector_size=SECTOR_SIZE,
+            sector_count=image_sectors,
+        )
+
+        # Create partition layout
+        gptutil.create_gpt_layout(
+            imagepath=disk_image_file,
+            sector_size=SECTOR_SIZE,
+            layout=volume,
+        )
```

Comment:
This seems like something we'd want to put in its own helper function, both for the ability to test this section on its own and so it's easier to replace if we ended up with something like...

```python
if we_want_gpt():
    create_gpt_image(disk_image_file, volume)
else:
    create_fancy_futuristic_image(disk_image_file, volume)
```